pkg/k8s/watchers/node.go

```
@@ -118,9 +123,58 @@ func (k *K8sWatcher) updateK8sNodeV1(oldK8sNode, newK8sNode *v1.Node) error {
 	if err != nil {
 		return err
 	}
+
+	k.updateCiliumNodeLabels(newK8sNode.Name, newNodeLabels)
+
 	return nil
 }
 
+func (k *K8sWatcher) updateCiliumNodeLabels(nodeName string, newNodeLabels map[string]string) {
+	var (
+		controllerName = fmt.Sprintf("sync-node-labels-with-ciliumnode (%v)", nodeName)
+		scopedLog      = log.WithFields(logrus.Fields{
+			logfields.Controller: controllerName,
+			logfields.Node:       nodeName,
+		})
+	)
+
+	k8sCM.UpdateController(controllerName,
+		controller.ControllerParams{
+			DoFunc: func(ctx context.Context) (err error) {
+				pName := &v1.Node{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: nodeName,
+					},
+				}
+
+				if k.ciliumNodeStore == nil {
+					return
```

We should return an error for this case because we want the controller to keep re-trying until the `ciliumNodeStore` is set.